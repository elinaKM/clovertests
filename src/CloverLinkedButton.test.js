import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Link } from "react-router-dom";

import {CloverLinkedButton} from 'cloveruicomponents';

const secondary = "#272121";
const primary = "yellow";
const bgText = "#f6e9e9";
const positive = "#BAEDBF";
const negative = "#FCA6A6";

var myFixedTheme = {
    palette: {
    body: "#363333",
    text: bgText,
    primary,
    primaryText: bgText,
    secondary,
    secondaryText: secondary,
    inverse: bgText,
    inverseText: "#363333",
    positive,
    negative,
    dark: "rgb(54, 51, 51)"
    }
}

describe('CloverLinkedButton', () => {
  it('renders correctly with all properties set, size=large', () => {
    const tree = renderer
      .create(
      <Router>
          <CloverLinkedButton size="large" to="https://www.google.com/" useTheme={myFixedTheme}>next</CloverLinkedButton>
      </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('CloverLinkedButton', () => {
it('renders correctly with all properties set, size=normal', () => {
    const tree = renderer
      .create(
      <Router>
          <CloverLinkedButton size="normal" to="https://www.google.com/" useTheme={myFixedTheme}>next</CloverLinkedButton>
      </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('CloverLinkedButton', () => {
  it('renders correctly with only required properties set', () => {
    const tree = renderer
      .create(
        <Router>
            <CloverLinkedButton to="https://www.google.com/">Home</CloverLinkedButton>
        </Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('CloverToggleButton', () => {
  it('renders correctly with no title given', () => {
    const tree = renderer
      .create(<Router>
          <CloverLinkedButton to="https://www.google.com/"></CloverLinkedButton>
          </Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});