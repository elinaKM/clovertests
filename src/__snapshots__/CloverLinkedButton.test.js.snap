// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloverLinkedButton renders correctly with all properties set, size=large 1`] = `
<a
  buttonSize="large"
  className="sc-bdVaJa fUsqPR"
  href="/https://www.google.com/"
  onClick={[Function]}
  themeProp={
    Object {
      "palette": Object {
        "body": "#363333",
        "dark": "rgb(54, 51, 51)",
        "inverse": "#f6e9e9",
        "inverseText": "#363333",
        "negative": "#FCA6A6",
        "positive": "#BAEDBF",
        "primary": "yellow",
        "primaryText": "#f6e9e9",
        "secondary": "#272121",
        "secondaryText": "#272121",
        "text": "#f6e9e9",
      },
    }
  }
>
  next
</a>
`;

exports[`CloverLinkedButton renders correctly with all properties set, size=normal 1`] = `
<a
  buttonSize="normal"
  className="sc-bdVaJa hKCCtf"
  href="/https://www.google.com/"
  onClick={[Function]}
  themeProp={
    Object {
      "palette": Object {
        "body": "#363333",
        "dark": "rgb(54, 51, 51)",
        "inverse": "#f6e9e9",
        "inverseText": "#363333",
        "negative": "#FCA6A6",
        "positive": "#BAEDBF",
        "primary": "yellow",
        "primaryText": "#f6e9e9",
        "secondary": "#272121",
        "secondaryText": "#272121",
        "text": "#f6e9e9",
      },
    }
  }
>
  next
</a>
`;

exports[`CloverLinkedButton renders correctly with only required properties set 1`] = `
<a
  buttonSize={undefined}
  className="sc-bdVaJa bYoCZr"
  href="/https://www.google.com/"
  onClick={[Function]}
  themeProp={
    Object {
      "fonts": Object {
        "code": "source-code-pro, Menlo, Monaco, Consolas, \\"Courier New\\",
        monospace",
        "main": "-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\",
        \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\",
        sans-serif",
      },
      "palette": Object {
        "body": "#363333",
        "dark": "rgb(54, 51, 51)",
        "inverse": "#f6e9e9",
        "inverseText": "#363333",
        "negative": "#FCA6A6",
        "positive": "#BAEDBF",
        "primary": "#196222",
        "primaryText": "#f6e9e9",
        "secondary": "#272121",
        "secondaryText": "#272121",
        "text": "#f6e9e9",
      },
    }
  }
>
  Home
</a>
`;

exports[`CloverToggleButton renders correctly with no title given 1`] = `
<a
  buttonSize={undefined}
  className="sc-bdVaJa bYoCZr"
  href="/https://www.google.com/"
  onClick={[Function]}
  themeProp={
    Object {
      "fonts": Object {
        "code": "source-code-pro, Menlo, Monaco, Consolas, \\"Courier New\\",
        monospace",
        "main": "-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\",
        \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\",
        sans-serif",
      },
      "palette": Object {
        "body": "#363333",
        "dark": "rgb(54, 51, 51)",
        "inverse": "#f6e9e9",
        "inverseText": "#363333",
        "negative": "#FCA6A6",
        "positive": "#BAEDBF",
        "primary": "#196222",
        "primaryText": "#f6e9e9",
        "secondary": "#272121",
        "secondaryText": "#272121",
        "text": "#f6e9e9",
      },
    }
  }
/>
`;
